Assets/Scripts/Map/Map.cs

```diff
@@ -0,0 +1,57 @@
+using UnityEngine;
+
+namespace Assets.Scripts.Map
+{
+    public class Map
```

Comment:
```suggestion
    public class Map: IMapState
```
Кажется, что этот класс реализует интерфейс `IMapState`

Reply:
Я думал что класс Map содержит в себе текущее состояние, а не является состоянием. Так что интуитивно кажется странным, что класс, который должен хранить в себе состояние, сам является частным случаем состояния. Но возможно это норм.